src/index.py

```diff
@@ -55,10 +55,10 @@ def handler(event, lambda_context):
         log_error_with_stacktrace(e, "Exception caught in top-level handler")
         result = TransformationResult.ProcessingFailed
 
-    try:
-        context.sfm.push_sfm_to_cloudwatch()
-    except Exception as e:
-        log_error_with_stacktrace(e, "SelfMonitoring push to Cloudwatch failed")
+    # try:
+    #     context.sfm.push_sfm_to_cloudwatch()
+    # except Exception as e:
+    #     log_error_with_stacktrace(e, "SelfMonitoring push to Cloudwatch failed")
```

Comment:
You could have commented it out inside of sfm and we would have them logged at least. But will do for today.

```
    def push_sfm_to_cloudwatch(self):
        metrics = self._generate_metrics()
        # cloudwatch = boto3.client('cloudwatch')
        # try:
        #     for i in range(0, len(metrics), 20):
        #         metrics_batch = metrics[i:(i + 20)]
        #         cloudwatch.put_metric_data(MetricData=metrics_batch, Namespace='DT/LogsStreaming')
        # except Exception as e:
        print("Print metrics on SFM push failure: " + str(metrics))
            # raise e
```

Reply:
They are still not good to be logged I think. Because right now SFM batch is getting very large when lots of records are coming in one batch (you can see example in this ticket's comments). We will also fix this in a simple way but it's not there yet

Reply:
And they are only logged on sfm push failure I think (not sure right now)

Reply:
>And they are only logged on sfm push failure I think (not sure right now)
Yes, I've pasted an edited version above that would log it always.
> Because right now SFM batch is getting very large
Yes, I remember, that's why won't discuss it today. Just commented to have a snippet ready for tomorrow's discussions.